sgxs-tools/src/sgx_detect/tests/scaffold.rs

```diff
@@ -24,20 +24,31 @@ pub trait Name {
     fn name(&self) -> &'static str;
 }
 
+#[derive(Debug, PartialEq)]
+pub enum BuildType {
+    Generic,
+    EnclaveOSPreInstall,
+    EnclaveOSPostInstall,
+}
+
 pub trait Print: Name {
-    fn try_supported(&self) -> Option<Status> {
-        Some(self.supported())
+    fn try_supported(&self, build_type: &BuildType) -> Option<Status> {
+        Some(self.supported_for_config(build_type))
     }
 
     fn supported(&self) -> Status {
-        panic!("Unable to answer supported query for {}", self.name())
+        panic!("Unable to answer supported query for {}\n", self.name())
+    }
+
+    fn supported_for_config(&self, _build_type: &BuildType) -> Status {
```

Comment:
I think it makes more sense to merge this into `fn supported`, like you did for `try_supported`. I don't think ignoring `build_type` is a big deal for implementors of this trait.

Reply:
@jethrogb  There are several calls to `fn suported` in the code. A lot of the calls are in functions where `BuildType` is not a parameter. It would become a very big change and maybe not even possible in some cases to pass `BuildType` recursively to all those parent functions. Is there a better way to do so ?

Reply:
See https://github.com/fortanix/rust-sgx/pull/166#discussion_r353163256